gngns.go

```diff
@@ -0,0 +1,47 @@
+package nmea
+
+const (
+	// Prefix GNGNS prefix
+	PrefixGNGNS            = "GNGNS"
+	NoFixGNGNS             = "N"
+	AutonomousGNGNS        = "A"
+	DifferentialGNGNS      = "D"
+	PreciseGNGNS           = "P"
+	RealTimeKinematicGNGNS = "R"
+	FloatRTKGNGNS          = "F"
+	EstimatedGNGNS         = "E"
+	ManualGNGNS            = "M"
+	SimulatorGNGNS         = "S"
+)
+
+type GNGNS struct {
+	BaseSentence
+	Time       Time
+	Latitude   float64
+	Longitude  float64
+	Mode       []string
```

Comment:
What if we did `Mode` like this instead:

``` go
type ModeGNGNS string

func (m ModeGNGNS) String() string {
  return string(m)
}

func (m ModeGNGNS) IsNoFix() bool {
  return strings.Contains(string(m), "N")
}

func (m ModeGNGNS) IsAutonomous() bool {
  return strings.Contains(string(m), "A")
}

// etc ...
```

Reply:
That would certainly be simpler. However, the values are positional based on which constellation has the appropriate fix. Eg. My receiver gives me "AAN", which tells me GPS and Glonass are up, but Galileo is down. Perhaps a convenience method for a simple test would be in order, in addition to positional values.

Reply:
How about something like this? 

```go
func (g GNGNS) IsMode(modes ...string) bool {
        for _, m := range g.Mode {
                for _, md := range modes {
                        if m == md {
                                return true
                        }
                }
        }
        return false
}
```

Reply:
@bmurray ah that makes more sense. In that case, let's leave it the way it is.